nodedb.go

```diff
@@ -382,22 +382,28 @@ func (ndb *nodeDB) getRoots() (map[int64][]byte, error) {
 // SaveRoot creates an entry on disk for the given root, so that it can be
 // loaded later.
 func (ndb *nodeDB) SaveRoot(root *Node, version int64) error {
-	ndb.mtx.Lock()
-	defer ndb.mtx.Unlock()
-
 	if len(root.hash) == 0 {
 		cmn.PanicSanity("Hash should not be empty")
 	}
+	return ndb.saveRoot(root.hash, version)
+}
+
+// SaveEmptyRoot creates an entry on disk for an empty root.
+func (ndb *nodeDB) SaveEmptyRoot(version int64) error {
+	return ndb.saveRoot([]byte{}, version)
```

Comment:
I would just call ndb.SaveRoot() here and keep the mtx in the exposed method, unless I'm missing something.

Reply:
The exposed method takes a root node though, which we don't have here.

Reply:
ok.